src/components/PopupWithForm.js

```diff
@@ -31,6 +30,19 @@ class PopupWithForm extends Popup {
       input.value = data[input.name];
     });
   }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
```

Comment:
Needs correction: Here you should return the initial button text (you can save it in the constructor like you do that for the confirm popup)